src/py21cmfast/drivers/lightcone.py

```diff
@@ -0,0 +1,987 @@
+    coeval_callback_redshifts : list or int, optional
+        Redshifts for `coeval_callback` computation.
+        If list, computes the function on `node_redshifts` closest to the specified ones.
+        If positive integer, computes the function on every n-th redshift in `node_redshifts`.
+        Ignored in the case `coeval_callback is None`.
+    use_interp_perturb_field : bool, optional
+        Whether to use a single perturb field, at the lowest redshift of the lightcone,
+        to determine all spin temperature fields. If so, this field is interpolated in the
+        underlying C-code to the correct redshift. This is less accurate (and no more efficient),
+        but provides compatibility with older versions of 21cmFAST.
+    cleanup : bool, optional
+        A flag to specify whether the C routine cleans up its memory before returning.
+        Typically, if `spin_temperature` is called directly, you will want this to be
+        true, as if the next box to be calculate has different shape, errors will occur
+        if memory is not cleaned. Note that internally, this is set to False until the
+        last iteration.
+    minimize_memory_usage
+        If switched on, the routine will do all it can to minimize peak memory usage.
+        This will be at the cost of disk I/O and CPU time. Recommended to only set this
+        if you are running particularly large boxes, or have low RAM.
+    lightcone_filename
+        The filename to which to save the lightcone. The lightcone is returned in
+        memory, and can be saved manually later, but including this filename will
+        save the lightcone on each iteration, which can be helpful for checkpointing.
+    return_at_z
+        If given, evaluation of the lightcone will be stopped at the given redshift,
+        and the partial lightcone object will be returned. Lightcone evaluation can
+        continue if the returned lightcone is saved to file, and this file is passed
+        as `lightcone_filename`.
+    \*\*global_kwargs :
+        Any attributes for :class:`~py21cmfast.inputs.GlobalParams`. This will
+        *temporarily* set global attributes for the duration of the function. Note that
+        arguments will be treated as case-insensitive.
+
+    Returns
+    -------
+    lightcone : :class:`~py21cmfast.LightCone`
+        The lightcone object.
+    coeval_callback_output : list
+        Only if coeval_callback in not None.
+
+    Other Parameters
+    ----------------
+    regenerate, write, direc, random_seed
+        See docs of :func:`initial_conditions` for more information.
+    """
+    direc, regenerate, hooks = _get_config_options(direc, regenerate, write, hooks)
+
+    if perturbed_fields is not None and initial_conditions is None:
+        raise ValueError(
+            "If perturbed_fields are provided, initial_conditions must be provided"
+        )
+
+    if cosmo_params is None and initial_conditions is None:
```

Comment:
Yup -- previous version had a warning that in v4 it would be required. It still has a default, so those wanting to just have the "old" behavior don't need to do anything.

Reply:
At least, not anything drastic

Reply:
Ah. You're right. The previous default was:

```python
RectilinearLightconer.with_equal_cdist_slices(
    min_redshift=redshift,
    max_redshift=max_redshift,
    resolution=user_params.cell_size,
    cosmo=cosmo_params.cosmo,
    quantities=lightcone_quantities,
    get_los_velocity=not flag_options.APPLY_RSDS,
)
```

I think I wanted the user to be more explicit about how they wanted to actually "lightconify" the simulation, rather than just relying on some black-box thing. But also constructing this is not totally simple, so I wouldn't argue too hard against providing it as a default. On the whole, I'd argue for documentation that makes it clear how to construct one of these, and make the user do it -- at least then they make the informed choice between rectilinear and angular lightcones.

Reply:
I'm happy with making the user generate the lightconer